elftools/dwarf/compileunit.py

```diff
@@ -73,14 +79,81 @@ def get_top_DIE(self):
         """ Get the top DIE (which is either a DW_TAG_compile_unit or
             DW_TAG_partial_unit) of this CU
         """
-        return self._get_DIE(0)
+
+        # Note that a top DIE always has minimal offset, no bisect required.
+        if self._diemap and self._diemap[0] == self.cu_die_offset:
```

Comment:
This code implies `_diemap` could be non-empty but the `[0]` element is not at the current die offset, is that true?

I'd like the invariants to be really well documented

Reply:
Yeah, I think the structure here is a little roundabout -- it *should* be sufficient to just check whether `_diemap` is empty and, if so, parse and memoize the DIE at `cu_due_offset`. Will test.